### PR TITLE
Fix some clang analyzer findings

### PR DESCRIPTION
**Pull request by OlegHahm**
Fix some findings by clang-analyzer (see also #3468).

### CODE REVIEW

**Comment by gebart** [comment]
This is the kind of PR I was hoping for with #3468 :smile:

**Comment by OlegHahm** [comment]
Yes, that's nice indeed - although none of these findings seems to have a real impact.

File: sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c [hunk 1]
```diff
@@ -415,7 +415,7 @@ static int _marshall_ethernet(ng_netdev_eth_t *dev, uint8_t *buffer, ng_pktsnip_
     if (netif_hdr->flags & NG_NETIF_HDR_FLAGS_BROADCAST) {
         _addr_set_broadcast(hdr->dst);
     }
-    else if (netif_hdr->flags & NG_NETIF_HDR_FLAGS_MULTICAST) {
+    else if ((netif_hdr->flags & NG_NETIF_HDR_FLAGS_MULTICAST) && (payload != NULL)) {
```
Comment (authmillenon) [inline sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c]: Sneaky way to reintroduce https://github.com/RIOT-OS/RIOT/pull/3238?

File: sys/net/network_layer/ng_icmpv6/ng_icmpv6.c [hunk 2]
```diff
@@ -62,10 +62,17 @@ void ng_icmpv6_demux(kernel_pid_t iface, ng_pktsnip_t *pkt)
 
     LL_SEARCH_SCALAR(pkt, icmpv6, type, NG_NETTYPE_ICMPV6);
 
+    if (icmpv6 == NULL) {
```
Comment (authmillenon) [inline sys/net/network_layer/ng_icmpv6/ng_icmpv6.c]: This should also be in here (otherwise the function isn't called) => use `assert()` here too

**Comment by authmillenon** [comment]
Apart from my two comments I'm fine with this PR.

File: sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c (hunk 1 above)
Comment (OlegHahm) [inline sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c]: Sorry, that wasn't intended. ;) Are you okay with the way it is done in the squash commit?

File: sys/net/network_layer/ng_icmpv6/ng_icmpv6.c (hunk 2 above)
Comment (OlegHahm) [inline sys/net/network_layer/ng_icmpv6/ng_icmpv6.c]: addressed

File: sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c [hunk 3]
```diff
@@ -415,7 +417,7 @@ static int _marshall_ethernet(ng_netdev_eth_t *dev, uint8_t *buffer, ng_pktsnip_
     if (netif_hdr->flags & NG_NETIF_HDR_FLAGS_BROADCAST) {
         _addr_set_broadcast(hdr->dst);
     }
-    else if (netif_hdr->flags & NG_NETIF_HDR_FLAGS_MULTICAST) {
+    else if (netif_hdr->flags & NG_NETIF_HDR_FLAGS_MULTICAST) && ) {
```
Comment (Kijewski) [inline sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c]: looks broken :)

File: sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c [hunk 4]
```diff
@@ -237,6 +237,9 @@ static void _send(ng_pktsnip_t *pkt)
         payload_len++;
     }
 #else
+    /* supporess clang-analyzer report about iface being not read */
```
Comment (Kijewski) [inline sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c]: s/supporess/suppress/

File: sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c (hunk 3 above)
Comment (OlegHahm) [inline sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c]: even syntax highlighting in vim agrees - I wonder if the PLL for my brain is working properly today.

**Comment by OlegHahm** [comment]
Ready to squash?

File: sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c (hunk 1 above)
Comment (authmillenon) [inline sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c]: Yes :-)

**Comment by authmillenon** [comment]
ACK, please squash.

**Pull request merged**